src/components/userProfile/CareerRow.js

```diff
@@ -10,73 +11,111 @@ class CareerRow extends Component {
+        loaded: true,
+      });
+    } else if (this.props.sectionID === 'Education') {
+      this.setState({
+        schoolName: this.getSchoolName(),
+        major: this.getMajor(),
+        graduationYear: this.graduationYear(),
+        loaded: true,
+      });
+    }
+  }
 
+  getCompanyName() {
     if (this.props.dataSource.employer) {
-      companyName = this.props.dataSource.employer.name;
-      identifier = 1;
-      if (this.props.dataSource.position) {
-        position = '| ' + this.props.dataSource.position.name;
-      }
+      return this.props.dataSource.employer.name;
+    }
 
-      if (this.props.dataSource.start_date) {
-        period = this.props.dataSource.start_date + ' - ' + this.props.dataSource.end_date;
-      }
-    } else if (this.props.dataSource.school) {
-      schoolName = this.props.dataSource.school.name;
-      identifier = 2;
-      if (this.props.dataSource.concentration.length > 0) {
-        major = this.props.dataSource.concentration[0].name;
-      }
+    return this.state.companyName;
+  }
 
-      if (this.props.dataSource.year) {
-        period = 'Class of ' + this.props.dataSource.year.name;
-      }
+  getPosition() {
+    if (this.props.dataSource.position) {
+      return this.props.dataSource.position.name;
     }
 
-    this.setState({
-      schoolName: schoolName,
-      major: major,
-      companyName: companyName,
-      position: position,
-      period: period,
-      identifier: identifier,
-      loaded: true,
-    });
+    return this.state.position;
   }
 
-  renderEducation() {
+  getPeriod() {
+    if (this.props.dataSource.start_date) {
+      return this.props.dataSource.start_date + ' - ' + this.props.dataSource.end_date;
+    }
+
+    return this.state.period;
+  }
+
+  getSchoolName() {
+    if (this.props.dataSource.school) {
+      return this.props.dataSource.school.name;
+    }
+
+    return this.state.schoolName;;
+  }
+
+  getMajor() {
+    if (this.props.dataSource.concentration.length > 0) {
+      return this.props.dataSource.concentration[0].name;
+    }
+
+    return this.state.major;
+  }
+
+  graduationYear() {
+    if (this.props.dataSource.year && this.props.dataSource.year.name !== '') {
+      return 'Class of ' + this.props.dataSource.year.name;
+    }
+
+    return this.state.graduationYear;;
+  }
+
+  renderExperience() {
     return (
       <View style={styles.container}>
-        <Text style ={styles.name}>{this.state.schoolName}</Text>
-        <Text style ={styles.name}>{this.state.major}</Text>
-        <Text style ={styles.period}>{this.state.period}</Text>
+        {this.state.position !== undefined &&
+          <Text style ={styles.name}>{this.state.position}</Text>
+        }
+        {this.state.companyName !== undefined &&
+          <Text style ={styles.name}>{this.state.companyName}</Text>
+        }
+        {this.state.period !== undefined &&
+          <Text style ={styles.period}>{this.state.period}</Text>
+        }
```

Comment:
Interesting code here!
You guys can check some information about this in [here](https://facebook.github.io/react/docs/conditional-rendering.html)